### PR TITLE
Add IqZone bidder

### PR DESCRIPTION
**Pull request by laarchenko**
https://github.com/prebid/prebid-server/pull/1964

### CODE REVIEW

**Review by And1sS — APPROVED** [review]

File: src/main/java/org/prebid/server/bidder/iqzone/IqzoneBidder.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                .flatMap(Collection::stream)
+                .map(bid -> constructBidderBid(bid, bidResponse, bidRequest))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private BidderBid constructBidderBid(Bid bid, BidResponse bidResponse, BidRequest bidRequest) {
+        try {
+            return BidderBid.of(bid, getBidType(bid.getImpid(), bidRequest.getImp()), bidResponse.getCur());
+        } catch (IllegalArgumentException e) {
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/iqzone/IqzoneBidder.java:80]: Seems this catch block can have only PreBidEx which can come from getBidType(), so what is the purpose of `catch (IllegalArgumentException e)`? You can see that there is no test for this case.

File: src/main/java/org/prebid/server/bidder/iqzone/IqzoneBidder.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    private List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> constructBidderBid(bid, bidResponse, bidRequest))
+                .filter(Objects::nonNull)
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/iqzone/IqzoneBidder.java:73]: ConstructBidderBid never returns null, pls remove this filter

File: src/main/java/org/prebid/server/bidder/iqzone/IqzoneBidder.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse), Collections.emptyList());
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            throw new PreBidException("Bid response is null or seat bid is empty");
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/iqzone/IqzoneBidder.java:61]: There is no error like this in go implementation

File: src/test/java/org/prebid/server/bidder/iqzone/IqzoneBidderTest.java [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    @Before
+    public void setUp() {
+        iqZoneBidder = new IqzoneBidder(ENDPOINT_URL, jacksonMapper);
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new IqzoneBidder("invalid_url", jacksonMapper));
+    }
+
```
Comment (snahornyi) [inline src/test/java/org/prebid/server/bidder/iqzone/IqzoneBidderTest.java:52]: Pls create test for makeHttp request method

File: src/main/resources/bidder-config/iqzone.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+adapters:
+  iqzone:
+    enabled: false
+    endpoint: http://smartssp-us-east.iqzone.com/pserver
+    pbs-enforces-gdpr: true
+    pbs-enforces-ccpa: true
+    modifying-vast-xml-allowed: false
```
Comment (snahornyi) [inline src/main/resources/bidder-config/iqzone.yaml:7]: Pls merge master and use adapter-defaults feature

**Review by snahornyi — CHANGES_REQUESTED** [review]

**Review by snahornyi — DISMISSED** [review]

**Review by rpanchyk — APPROVED** [review]

**Pull request merged**